packages/communication-react/review/communication-react.api.md

```diff
@@ -2042,6 +2042,8 @@ export const SendBox: (props: SendBoxProps) => JSX.Element;
 export interface SendBoxProps {
     autoFocus?: 'sendBoxTextField' | false;
     disabled?: boolean;
+    // @beta
+    onRenderAttachedFiles?: () => JSX.Element;
```

Comment:
## The Big Question ;)

Should this be a render callback (i.e., a callback that returns `JSX.Element`)?

I feel that a more natural API would be:

```ts
  export interface SendBoxProps {
  ...
  activeFileUploads?: FileUploadState[]
  // This to _override_ how the file uploads are displayed.
  onRenderFileUploads?: (activeUploads: FileUploadState[]) => JSX.Element;
```

That is:

* Showing file uploads is an integral part of the `SendBox`. By default, it uses the `FileCardGroup` to do this.
* `onRenderFileUpload` is then a way for Contoso to override the default rendering.

This is consistent with how other components render nested components.

This would mean that we need to move `FileCardGroup` and its dependencies to `packges/react-components`, but we don't have to export it as public API.

Reply:
Also, this `onRender` conforms to the `onRender` functions sprinkled around our library.
Also, I am having a hard time understanding the use of `activeFileUploads` as a separate prop if we have `onRenderFileUploads?: (activeUploads: FileUploadState[]) => JSX.Element;`

Reply:
I'm confused by @anjulgarg comments above.

1. My suggestion is intended to make it _easier_ for Contoso to use just the `SendBox` component. They get the file upload UI without having to supply render functions. Currently, they need to implement their own `FileCard` et al to be able to use this.
2. The `onRender*` props override default behaviour elsewhere, same as I'm suggesting here.

![image](https://user-images.githubusercontent.com/82062616/153645911-d9dcba9d-1369-43d5-a2bb-79e801209a08.png)
Each of those `onRender*` props [_overrides_ default behavior](https://github.com/Azure/communication-ui-library/blob/main/packages/react-components/src/components/MessageThread.tsx#L465)

![image](https://user-images.githubusercontent.com/82062616/153646135-ca84d756-2860-4090-a603-95ad04ee4b01.png)

Same here, the `onRender*` props [_override_ default behavior](https://github.com/Azure/communication-ui-library/blob/main/packages/react-components/src/components/ParticipantList.tsx#L61).

Reply:
IRL discussion:

[Create another task]
* Add a prop for `activeFileUploads`
* Move `FileCardGroup` to a `@internal` component in `react-components`, and use it here.
* Change the `onRender` function to be an override, that gets the required params (e.g. MessageThread.onRenderMessageStatus() etc.)

Lots of discussion points:
* How would Contoso use the `activeFileUploads` prop
* Should the `onRender` override also provide a defaultRender method (that uses internal FileCardGroup)
* What about errors from file uploads (still open, whether this is part of SendBox), probably a controlled prop from Contoso to set the initial error.

Reply:
Let's make this discussion thread non-blocking.